dpgen2/conf/conf_generator.py

```diff
@@ -0,0 +1,95 @@
+import dpdata
+import dargs
+import tempfile
+from pathlib import Path
+from typing import (
+    List, Dict,
+)
+from abc import (
+    ABC, abstractmethod,
+)
+
+class ConfGenerator(ABC):
+    @abstractmethod
+    def generate(
+            self,
+            type_map,            
+    ) -> dpdata.MultiSystems:
+        r"""Method of generating configurations.
+
+        Parameters
+        ----------
+        type_map: List[str]
+                The type map.
+
+        Returns
+        -------
+        confs:  dpdata.MultiSystems
+                The returned configurations in `dpdata.MultiSystems` format
+        
+        """
+        pass
+
+
+    def get_file_content(
+            self, 
+            type_map,
+            fmt='lammps/lmp',
+    ) -> List[str]:
+        r"""Get the file content of configurations
+
+        Parameters
+        ----------
+        type_map: List[str]
+                The type map.
+
+        Returns
+        -------
+        conf_list: List[str]
+                A list of file content of configurations.
+
+        """
+        ret = []
+        ms = self.generate(type_map)
+        for ii in range(len(ms)):
+            ss = ms[ii]
+            for jj in range(ss.get_nframes()):
+                tf = Path(tempfile.NamedTemporaryFile().name)
+                ss[jj].to(fmt, tf)
+                ret.append(tf.read_text())
+                tf.unlink()
```

Comment:
`tempfile.NamedTemporaryFile` returns a file-like object. Using in this way will make the file not closed even if it is deleted.

It makes sense to use a `with` statement.
```suggestion
                with tempfile.NamedTemporaryFile() as ft:
                    tf = Path(ft.name)
                    ss[jj].to(fmt, tf)
                    ret.append(tf.read_text())
```